src/Syntax_Client/Oni_Syntax_Client.re

```diff
@@ -0,0 +1,114 @@
+/*
+ Syntax client
+ */
+
+module Core = Oni_Core;
+
+open Oni_Syntax;
+
+module LogClient = (val Core.Log.withNamespace("Syntax.client"));
+module LogServer = (val Core.Log.withNamespace("Syntax.server"));
+
+type t = {
+  in_channel: Stdlib.in_channel,
+  out_channel: Stdlib.out_channel,
+  readThread: Thread.t,
+  //writeThread: Thread.t,
+};
+
+let write = (v: t, msg: Protocol.ClientToServer.t) => {
+  Marshal.to_channel(v.out_channel, msg, []);
+  Stdlib.flush(v.out_channel);
+};
+
+exception SyntaxProcessCrashed;
+
+let start = languageInfo => {
+  let (pstdin, stdin) = Unix.pipe();
+  let (stdout, pstdout) = Unix.pipe();
+  let (stderr, pstderr) = Unix.pipe();
+
+  Unix.set_close_on_exec(pstdin);
+  Unix.set_close_on_exec(stdin);
+  Unix.set_close_on_exec(pstdout);
+  Unix.set_close_on_exec(stdout);
+  Unix.set_close_on_exec(pstderr);
+  Unix.set_close_on_exec(stderr);
+
+  let pid =
+    Unix.create_process(
+      Sys.executable_name,
+      [|Sys.executable_name, "--syntax-highlight-service"|],
+      pstdin,
+      pstdout,
+      pstderr,
+    );
+
+  let shouldClose = ref(false);
+
+  let in_channel = Unix.in_channel_of_descr(stdout);
+  let out_channel = Unix.out_channel_of_descr(stdin);
+
+  let waitThread =
+    Thread.create(
+      () => {
+        let _ = Unix.waitpid([], pid);
```

Comment:
Ignoring values without type annotations produce a warning in 4.08.

Reply:
Good point - I'm pulling the return code now, and still ignoring the status, but annotated it.